src/net/sourceforge/kolmafia/Modifiers.java

```diff
@@ -3628,80 +3733,82 @@ else if (itemId != ItemPool.TOILET_PAPER) {
-    Modifiers.writeModifierCategory(writer, campground, "Campground", "Campground equipment");
+    Modifiers.writeModifierCategory(
+        writer, campground, ModifierType.CAMPGROUND, "Campground equipment");
     writer.println();
-    Modifiers.writeModifierCategory(writer, effects, "Effect", "Status Effects");
+    Modifiers.writeModifierCategory(writer, effects, ModifierType.EFFECT, "Status Effects");
     writer.println();
-    Modifiers.writeModifierCategory(writer, passives, "Skill", "Passive Skills");
+    Modifiers.writeModifierCategory(writer, passives, ModifierType.SKILL, "Passive Skills");
     writer.println();
-    Modifiers.writeModifierCategory(writer, outfits, "Outfit", "Outfits");
+    Modifiers.writeModifierCategory(writer, outfits, ModifierType.OUTFIT, "Outfits");
     writer.println();
-    Modifiers.writeModifierCategory(writer, zodiacs, "Sign", "Zodiac Sign");
+    Modifiers.writeModifierCategory(writer, zodiacs, ModifierType.SIGN, "Zodiac Sign");
     writer.println();
-    Modifiers.writeModifierCategory(writer, statdays, "StatDay", "Stat Day");
+    Modifiers.writeModifierCategory(writer, statdays, ModifierType.EVENT, "Stat Day");
     writer.println();
-    Modifiers.writeModifierCategory(writer, zones, "Zone", "Zone-specific");
+    Modifiers.writeModifierCategory(writer, zones, ModifierType.ZONE, "Zone-specific");
     writer.println();
-    Modifiers.writeModifierCategory(writer, locations, "Loc", "Location-specific");
+    Modifiers.writeModifierCategory(writer, locations, ModifierType.LOC, "Location-specific");
     writer.println();
-    Modifiers.writeModifierCategory(writer, synergies, "Synergy", "Synergies");
+    Modifiers.writeModifierCategory(writer, synergies, ModifierType.SYNERGY, "Synergies");
     writer.println();
-    Modifiers.writeModifierCategory(writer, mutexes, "Mutex", "Mutual exclusions");
+    Modifiers.writeModifierCategory(
+        writer, mutexes, ModifierType.MUTEX_GENERIC, "Mutual exclusions");
     writer.println();
-    Modifiers.writeModifierCategory(writer, maximization, "MaxCat", "Maximization categories");
+    Modifiers.writeModifierCategory(
+        writer, maximization, ModifierType.MAX_CAT, "Maximization categories");
     writer.println();
-    Modifiers.writeModifierCategory(writer, potions, "Item", "Everything Else");
-    Modifiers.writeModifierCategory(writer, freepulls, "Item");
-    Modifiers.writeModifierCategory(writer, wikiname, "Item");
+    Modifiers.writeModifierCategory(writer, potions, ModifierType.ITEM, "Everything Else");
+    Modifiers.writeModifierCategory(writer, freepulls, ModifierType.ITEM);
+    Modifiers.writeModifierCategory(writer, wikiname, ModifierType.ITEM);
 
     writer.close();
   }
 
   private static void writeModifierCategory(
-      final PrintStream writer, final Set<String> set, final String type, final String tag) {
+      final PrintStream writer, final Set<String> set, final ModifierType type, final String tag) {
```

Comment:
a note to me or you to check before merge: there's a function somewhere that writes out `modifiers.txt`, and it would be nice to confirm it still does what it used to do

Reply:
I started down the road of adding a test to ensure that `writeModifiers` is writing out a subset of the lines in `modifiers.txt`, but it turns out that there are a number of ways in which it's not arranged exactly the same way. For example, the Spelunky items are all in their own section and not in the Hats/Weapons/whatever sections. And there are quite a few slight sorting errors where things are misalphabetized. We could correct this but I don't think it's a priority. With my latest commit I've verified that `writeModifiers` does essentially the right thing.

Reply:
Added a disabled test for this.

Reply:
ace!